pa11ycrawler/html.py

```diff
@@ -38,117 +44,115 @@ def main():
     args = parser.parse_args()
-    data_dir = os.path.expanduser(args.data_dir)
-    if not os.path.exists(data_dir):
+    data_dir = Path(args.data_dir).expand()
+    if not data_dir.isdir():
         msg = "Data directory {dir} does not exist".format(dir=args.data_dir)
         raise ValueError(msg)
-    data_filenames = [name for name in os.listdir(data_dir)
+    data_filenames = [name for name in data_dir.files()
                       if name.endswith(".json")]
     if not data_filenames:
         msg = "Data directory {dir} contains no JSON files".format(dir=args.data_dir)
         raise ValueError(msg)
-    output_dir = os.path.expanduser(args.output_dir)
-    if not os.path.exists(output_dir):
-        os.makedirs(output_dir)
+    output_dir = Path(args.output_dir).expand()
+    output_dir.makedirs_p()
 
     return render_html(data_dir, output_dir)
 
 
-COLOR_CLASSES = {
-    'error': 'danger',
-    'warning': 'warning',
-    'notice': 'info',
-}
-
-
-def get_code_info(code):
+def wcag_refs(code):
     """
-    Method for getting links to docs to present in reports.
-
-    Args: code (example: WCAG2AA.Principle3.Guideline3_2.3_2_1.G107)
-    Returns: a dict mapping technique to documentation url.
+    Given a `code` from pa11y, return a list of the WCAG references.
+    These references are always of the form: one or more capital letters,
+    followed by one or more digits. One `code` may contain multiple
+    references, separated by commas.
     """
-    link_info = {'doc_links': {}, 'base_code': ''}
-    base_url = 'https://www.w3.org/TR/WCAG20-TECHS/'
+    bits = code.split(".")
+    for bit in reversed(bits):
+        if WCAG_REFS_RE.match(bit):
+            return bit.split(",")
+    return []
 
-    guide_pattern = re.compile(r'WCAG2AA.Principle\d.Guideline[0-9\_\.]*')
-    base_code_matches = guide_pattern.match(code)
 
-    if not base_code_matches:
-        log.debug(
-            'Code {code} doesn\'t match expected pattern. Unable to produce '
-            'documentation links for reports.'.format(code=code)
+def copy_assets(output_dir):
+    """
+    Copy static assets (CSS, JS, fonts, etc) into the output directory.
+    """
+    node_modules = REPO_DIR / "node_modules"
+    if not node_modules.isdir():
+        msg = (
+            "Node modules have not been installed, cannot copy assets. "
+            "To install them, run `npm install`."
         )
-        return link_info
+        raise RuntimeError(msg)
 
-    link_info['base_code'] = base_code_matches.group()[0:-1]
-    split_code = guide_pattern.split(code)
-    tech_group = split_code[1]
-    tech_pattern = re.compile(r'[A-Z]+\d{1,3}')
-    techs = tech_pattern.findall(tech_group)
+    assets_dir = output_dir / 'assets'
 
-    for tech in techs:
-        link_info['doc_links'][tech] = base_url + tech
-
-    return link_info
+    # bootstrap has a well-organized `dist` directory
+    (node_modules / 'bootstrap' / 'dist').merge_tree(assets_dir)
```

Comment:
I don't know `merge_tree`. What does that do?

Perhaps related, do you need to `mkdirs_p` the `assets_dir`?